typescript/agentkit/src/action-providers/cdp/cdpSqlApiDescription.ts

```diff
@@ -0,0 +1,384 @@
+        type: "String",
+        description: "Transaction recipient (EOA or contract)",
+      },
+      { fieldName: "transaction_from", type: "String", description: "Transaction sender (EOA)" },
+      {
+        fieldName: "log_index",
+        type: "uint32",
+        description: "Log index within the transaction (0-based)",
+      },
+      {
+        fieldName: "address",
+        type: "String",
+        description: "Contract address that created the log",
+      },
+      {
+        fieldName: "topics",
+        type: "Array(String)",
+        description: "Indexed params / signature hash",
+      },
+      {
+        fieldName: "action",
+        type: "Enum8('removed' = -1, 'added' = 1)",
+        description: "1 = created; −1 = reorged out; sum > 0 means active",
+      },
+    ],
+  },
+  // not public yet
+  // {
+  //   tableName: "base.transfers",
+  //   fields: [
+  //     {
+  //       fieldName: "block_number",
+  //       type: "uint64",
+  //       description: "Block number containing the transfer",
+  //     },
+  //     { fieldName: "block_timestamp", type: "DateTime64", description: "Block timestamp" },
+  //     { fieldName: "transaction_to", type: "String", description: "Transaction recipient address" },
+  //     { fieldName: "transaction_from", type: "String", description: "Transaction sender address" },
+  //     { fieldName: "log_index", type: "uint32", description: "Log index within the transaction" },
+  //     { fieldName: "token_address", type: "String", description: "Token contract address" },
+  //     {
+  //       fieldName: "from_address",
+  //       type: "String",
+  //       description: "Address tokens were transferred from",
+  //     },
+  //     {
+  //       fieldName: "to_address",
+  //       type: "String",
+  //       description: "Address tokens were transferred to",
+  //     },
+  //     { fieldName: "value", type: "uint256", description: "Amount of tokens transferred" },
+  //     { fieldName: "action", type: "Enum8", description: "Action flag: 1 add, −1 reorg removal" },
+  //   ],
+  // },
+];
+
+const schemaJson = JSON.stringify(apiSchema, null, 2);
+
+export const description = `
+    This action executes read-only SQL queries against indexed blockchain data using the CDP SQL API.
+    
+    **Use Cases:**
+    - Query transaction history and patterns
+    - Analyze event logs and smart contract interactions
+    - Retrieve block information and metadata
+    - Examine token transfers and DeFi activity
+    
+    **IMPORTANT Query Requirements:**
+    - Must be SELECT statements only (ClickHouse SQL dialect)
+    - Casts use the ::<Type> syntax (not CAST(... AS ...))
+    - Maximum query length: 10,000 characters
+    - Maximum result rows: 10,000
+    - Query timeout: 30 seconds
+    - Maximum JOINs: 5
+    - No cartesian products allowed
+    - No DDL/DML operations (INSERT, UPDATE, DELETE, etc.)
+    - Keep it simple and break down the task into several queries if appropriate.
```

Comment:
Please add this line:
```suggestion
    - No DDL/DML operations (INSERT, UPDATE, DELETE, etc.)
    - Always normalize addresses with lower('<addr>')
    - Keep it simple and break down the task into several queries if appropriate.
```